packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -416,7 +429,7 @@ class InputDecorator extends StatelessWidget {
       decoration: new BoxDecoration(
         border: new Border(
           bottom: new BorderSide(
-            color: borderColor,
+            color: hideUnderline ? Colors.transparent : borderColor,
```

Comment:
ok i'll use BorderStyle.none for hide underline. in my project i need Text Input with placeholder and custom underline